version: "2.1"

services:
  todo-postgres:
    image: todo-postgres
    container_name: todo-postgres
    ports:
      - 5432:5432
    restart: on-failure
    build: ./env/PostgreSQL
    environment:
      - POSTGRES_HOST_AUTH_METHOD=trust
      - POSTGRES_PASSWORD=postgres
    volumes:
      - ./env/PostgreSQL/init.sql:/docker-entrypoint-initdb.d/init.sql
    logging:
      driver: "json-file"
      options:
        max-file: "5"
        max-size: "100m"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 15
    networks:
      - todo-network

  todo-pgadmin:
    image: todo-pgadmin
    container_name: todo-pgadmin
    restart: on-failure
    build: ./env/pgAdmin
    environment:
      - PGADMIN_DEFAULT_EMAIL=admin
      - PGADMIN_DEFAULT_PASSWORD=admin
    depends_on:
      todo-postgres:
        condition: service_healthy
    ports:
      - 5050:80
    networks:
      - todo-network

  seq:
    image: datalust/seq:latest
    restart: on-failure
    container_name: seq
    environment:
      - ACCEPT_EULA=Y
    ports:
      - 5341:80
    networks:
      - todo-network

networks:
  todo-network:
    name: todo-network
